tools/mason/MasonPublish.chpl

```diff
@@ -0,0 +1,248 @@
+/*
+ * Copyright 2004-2019 Cray Inc.
+ * Other additional copyright holders may be indicated within.
+ *
+ * The entirety of this work is licensed under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ *
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+
+use MasonUtils;
+use Spawn;
+use FileSystem;
+use TOML;
+use MasonEnv;
+use MasonNew;
+use MasonModify;
+use Random;
+
+/* Top Level procedure that gets called from mason.chpl that takes in arguments from command line
+Returns the help output in '-h' or '--help' exits in the arguments.
+If --dry-run is passed then it checks to see if the package is able to be published.
+Takes in the username of the package owner as an argument*/
+proc masonPublish(args: [] string) throws {
+  try! {
+    if args.size == 4 {
+      masonPublishHelp();
+      exit(0);
+    }
+    var dry = false;
+    var username = getUsername();
+    if username == '' {
+      throw new owned MasonError('Must have remote origin set up in repository to publish');
```

Comment:
If I run ``mason publish`` in the chapel repo I get the following uncaught error.

```chpl
a6dca568a00eca8b5290080a3bafc0984b21ed19	HEAD
616b234c620d45fa26efbaf6283c64cff935bba4	refs/heads/fix-docs
a6dca568a00eca8b5290080a3bafc0984b21ed19	refs/heads/master
uncaught PermissionError: Permission denied (in remove with path "/private/tmp/powerlog")
  $CHPL_HOME/modules/standard/SysError.chpl:416: thrown here
  ./MasonPublish.chpl:35: uncaught here
```
my guess is that this kind of user error will happen pretty frequently(git repo with remotes that isnt a mason package) so lets get a nice error message for that. (This is not required to merge imo)

Reply:
Well done on handling this, however, I got the following error message

```chpl
fatal: unable to access 'https://github/comSpartee/mason-registry/': Could not resolve host: github
Spartee is not a valid GitHub username or mason-registry is not forked
Make sure you are connected to the internet before publishing.
```

maybe need some reformatting there? given that spartee is a good username for me haha